traffic_ops/traffic_ops_golang/user/current.go

```diff
@@ -309,7 +316,16 @@ func ReplaceCurrent(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	if err = updateUser(&user, tx, changePasswd, changeConfirmPasswd); err != nil {
+	if useV4User {
+		userV4 = user.Upgrade()
+	}
+
+	if useV4User {
+		err = updateUserV4(userV4, inf.Tx)
```

Comment:
So, as far as I can tell, the difference between userV4 and user is really just `Role`, which isn't even used/updated in the `updateUser` function. So, why doesn't just calling `updateUser(&user, tx, changePasswd, changeConfirmPasswd)` actually work?

Reply:
Good catch, and actually I didn't want to use `UpdateQueryV40` anyways. Hold pls for an update.